lib/utils/job-utils/command-util.js

```diff
@@ -104,6 +106,65 @@ function commandUtilFactory(
         });
      };
 
+    CommandUtil.prototype.sshExec = function(cmdObj, sshSettings, sshClient) {
+        return new Promise(function(resolve, reject) {
+            if(cmdObj.timeout) {
+                setTimeout(function() {
+                    var seconds = cmdObj.timeout / 1000;
+                    reject(new Error('The remote operation timed out after '+
+                                 seconds + ' seconds'));
+                }, cmdObj.timeout);
+            }
+            var ssh = sshClient;
+            ssh.on('ready', function() {
+                ssh.exec(cmdObj.cmd, function(err, stream) {
+                    if (err) { reject(err); }
+                    stream.on('close', function(code) {
+                        cmdObj.exitCode = code;
+                        ssh.end();
+                    }).on('error', function(err) {
+                        reject(err);
+                    }).on('data', function(data) {
+                        cmdObj.stdout = ( cmdObj.stdout || '' ) + data.toString();
+                    }).stderr.on('data', function(data) {
+                        cmdObj.stderr = ( cmdObj.stderr || '' ) + data.toString();
+                    });
+                });
+            })
+            .on('error', function(err) {
+                logger.error('ssh error', {
+                    error: err,
+                    host: sshSettings.host,
+                    task: cmdObj,
+                    level: err.level,
+                    description: err.description
+                });
+                reject(err);
+            })
+            .on('close', function(hasErr) {
+
+                if (hasErr || (cmdObj.exitCode &&
+                    !_.contains(cmdObj.acceptedResponseCodes, cmdObj.exitCode))) {
+                    logger.error("Failure running remote command", {task:cmdObj});
+
+                    reject(new Error(
+                            "Encountered a failure running "+cmdObj.cmd+
+                            "on remote host"+ sshSettings.host
```

Comment:
you might consider appending in the nodeID to this error message as well, as it may not be easy to determine which Node has the relevant host settings.